### PR TITLE
Update tensorflow-notebook-image workflow to pin tf and cuda version

### LINKED ISSUE #375: Jupyter notebook image should pin TF version
The latest image for jupyter notebook uses tensorflow 1.5-dev version and the TFServing image uses tensorflow 1.4 release.

I ran into an issue where I trained a model using the jupyter notebook, but had compatibility issues while serving that model with TFServing.

We should try to maintain the same tensorflow versions in the various kubeflow images.

### LINKED ISSUE #467: Build Jupyter Notebook images for supported versions of TF
We should start building/maintaining multiple Jupyter images corresponding to different versions of TF. 

Related #375

### PR DESCRIPTION
**Pull request by ankushagarwal**
* Build 1.4.1, 1.5.1, 1.6.0 versions for gpu and cpu
* Update ksonnet version
* Update bazel version
* Built all images locally to test it.

The images will have the following URI format
```
gcr.io/kubeflow-images-staging/tensorflow-1.4.1-notebook-gpu:20180320-afw9aow
gcr.io/kubeflow-images-staging/tensorflow-1.4.1-notebook-cpu:20180320-afw9aow
```

Fixes #375

Fixes #467 (Multiple Jupyter images for different TF Versions).

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/471)

### CODE REVIEW

File: components/tensorflow-notebook-image/releaser/components/workflows.libsonnet [hunk 1]
```diff
@@ -124,10 +122,26 @@
         },
         sidecars: sidecars,
       };  // buildTemplate
-
-      local buildImageTemplate(step_name, image, base_image, tf_package) =
-        buildTemplate(
-          step_name,
+      local buildImageTemplate(tf_version, device) = {
+        local image = params.registry + "/tensorflow-notebook-" + device + ":" + tf_version + "-" + params.versionTag,
```
Comment (jlewi) [inline components/tensorflow-notebook-image/releaser/components/workflows.libsonnet]: nil: I think the tf_version should be in the image name not the tag
e.g
"tensorflow-notebook-"+ tf_version + "-" device + ":" + params.versionTag

Comment (ankushagarwal) [reply components/tensorflow-notebook-image/releaser/components/workflows.libsonnet]: Done

File: components/tensorflow-notebook-image/releaser/components/workflows.libsonnet [hunk 2]
```diff
@@ -124,10 +122,26 @@
         },
         sidecars: sidecars,
       };  // buildTemplate
-
-      local buildImageTemplate(step_name, image, base_image, tf_package) =
-        buildTemplate(
-          step_name,
+      local buildImageTemplate(tf_version, device) = {
+        local image = params.registry + "/tensorflow-notebook-" + tf_version + "-" + device + ":" + params.versionTag,
```
Comment (jlewi) [inline components/tensorflow-notebook-image/releaser/components/workflows.libsonnet]: nit. "tensorflow-" + tf_version + "-notebook"

I think this makes it more clear that tf_version applies to tensorflow

Comment (ankushagarwal) [reply components/tensorflow-notebook-image/releaser/components/workflows.libsonnet]: Done

**Pull request merged**